git-redate

```diff
@@ -1,209 +1,357 @@
-#!/bin/bash
-
-is_git_repo() {
-    git rev-parse --show-toplevel > /dev/null 2>&1
-    result=$?
-    if test $result != 0; then
-        >&2 echo 'Not a git repo!'
-        exit $result
-  fi
+#!/bin/sh
+
+dashless="${0##*/}"
+dashless="${dashless%%-*} ${dashless#*-}"
+
+# This variable is consulted by the code in "git-sh-setup", which is sourced
+# below -- we don't need to export the variable.
+# shellcheck disable=SC2034
+OPTIONS_SPEC="\
+${dashless} [<options>] [[--] <rev-list-options>]
+
+Change the dates of one or more git commits according to an interactively-specified scheme.
+
+${dashless} --all
+${dashless} --count 10
+${dashless} HEAD~10..HEAD~6
+--
+d,debug              show diagnostic output
+c,commits=<commits>  number of commits to re-date (default: 5)
+l,limit=<limit>      number of commits to re-date in a single batch (default: 20)
+a,all                re-date all commits
+n,dry-run            show re-dating commands, but do not execute them
+original=<namespace> namespace where original commits will be stored (default: refs/original)
+S,gpg-sign?<keyid>   sign commits (negate with --no-gpg-sign)
+no-gpg-sign*         do not sign commits
+"
+
+# Allow running this command from a subdirectory of the working tree
+# shellcheck disable=SC1091
+SUBDIRECTORY_OK=yes . git-sh-setup
```

Comment:
In order for this to work on my system I had to change this line:
```sh
SUBDIRECTORY_OK=yes . "$(git --exec-path)/git-sh-setup"
```
Taken from: https://git-scm.com/docs/git-sh-setup

Reply:
Thanks.  I assume you are invoking this via `git-redate` directly rather than [installing `git-redate` into a `$PATH` entry](https://github.com/PotatoLabs/git-redate#installation)?  In any case, I guess there's no harm in fully-qualifiying the `git-sh-setup` path rather than relying on a `$PATH` search that depends on the script being run as `git redate`.

Reply:
This is done.